src/jormungandr/explorer/src/api/graphql/mod.rs

```diff
@@ -1398,6 +1394,41 @@ impl VotePlanStatus {
     }
 }
 
+/// checks if the tally is public or private and converts result to TallyStatus type
+pub fn to_tally_status(tally: Option<ExplorerVoteTally>) -> Option<TallyStatus> {
+    tally.map(|tally| match tally {
+        ExplorerVoteTally::Public { results, options } => TallyStatus::Public(TallyPublicStatus {
+            results: results.iter().map(Into::into).collect(),
+            options: options.into(),
+        }),
+        ExplorerVoteTally::Private { results, options } => {
+            TallyStatus::Private(TallyPrivateStatus {
+                results: results.map(|res| res.iter().map(Into::into).collect()),
+                options: options.into(),
+            })
+        }
+    })
+}
+
+// if the tally is None, convert to generic tally result as per rest api requirements
+pub fn generic_tally_status(p: ExplorerVoteProposal) -> Option<TallyStatus> {
+    let s = StakeControl::default();
+    match compute_public_tally(&p, &s) {
+        ExplorerVoteTally::Public { results, options } => {
+            Some(TallyStatus::Public(TallyPublicStatus {
+                results: results.iter().map(Into::into).collect(),
+                options: options.into(),
+            }))
+        }
+        ExplorerVoteTally::Private { results, options } => {
+            Some(TallyStatus::Private(TallyPrivateStatus {
+                results: results.map(|res| res.iter().map(Into::into).collect()),
+                options: options.into(),
+            }))
+        }
+    }
+}
+
```

Comment:
Forgot to add this comment.

Could this return `TallyStatus` instead of `Option<TallyStatus>`? Because every branch returns `Some`

Reply:
yes good idea, thanks. Addressed.